### PR TITLE
[Merged by Bors] - Relax constraint on CA mandatory presence to comply with expected behavior in K8s

### LINKED ISSUE #255: Support for Rancher Kubernetes Engine (not Rancher Desktop)
RKE has one specificity, it adds a layer of Enterprise IAM and an Authentication Proxy with RBAC support. 
[RKE Authentication](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/manage-clusters/access-clusters/authorized-cluster-endpoint) does not need any CA to be configured in kube config client side. Instead it uses a generated long-lived OIDC Bearer token.

This is not compatible with this crate implementation as defined [here](https://github.com/infinyon/k8-api/blob/65c1b0ee9fc25d4260673bde1caa07baae55edfb/src/k8-client/src/cert.rs#L171) 

        let builder = if let Some(ca_data) = &current_cluster.cluster.certificate_authority_data {
            debug!("detected in-line cluster CA certs");
            let pem_bytes = decode(ca_data).unwrap();
            builder.load_ca_cert_with_data(pem_bytes)?
        } else {
            // let not inline, then must must ref to file
            let ca_certificate_path = current_cluster
                .cluster
                .certificate_authority
                .as_ref()
                .ok_or_else(|| {
                    IoError::new(
                        ErrorKind::InvalidInput,
                        "current cluster must have CA crt path".to_owned(),
                    )
                })?;

            debug!("loading cluster CA from: {:#?}", ca_certificate_path);

            builder.load_ca_certificate(ca_certificate_path)?
        };

I am willing to contribute the OIDC / Oauth2 compatible implementation but I just wanted to know if it was something you would consider as valuable somehow.

### PR DESCRIPTION
**Pull request by kikokikok**
resolves #255

### CODE REVIEW

File: src/k8-client/src/client/config_native.rs [hunk 1]
```diff
@@ -161,8 +161,9 @@ impl ConfigBuilder for HyperClientBuilder {
             None => ConnectorBuilder::anonymous(),
         };
 
-        if ca_cert.is_some(){
-            connector_builder = connector_builder.add_root_certificate(ca_cert.expect("failed while using server CA builder"))?
+        if ca_cert.is_some() {
+            connector_builder = connector_builder
+                .add_root_certificate(ca_cert.expect("failed while using server CA builder"))?
```
Comment (sehz) [inline src/k8-client/src/client/config_native.rs:166]: I know this was in previous code but can you remove expect?

**Review by sehz — CHANGES_REQUESTED** [review]
Looks good.  couple of minor nits.
In addition, should bump up crate version as well

File: src/k8-client/src/client/config_native.rs (hunk 1 above)
Comment (kikokikok) [reply src/k8-client/src/client/config_native.rs]: I removed it in both implementations Native and Openssl

**Review by sehz — APPROVED** [review]
LGTM.  Nice changes

**Pull request closed without merging**